src/decoder.rs

```diff
@@ -49,77 +49,22 @@ pub enum Instr {
     Mret,
     Sret,
     Vfencevma,
-    /// Load Double-Word (register-based)
-    Ld {
+    /// Load (register-based)
+    Load {
         rd: Register,
         rs1: Register,
         imm: isize,
+        len: usize,
+        compressed: bool,
+        unsigned: bool,
```

Comment:
A good convention to name boolean variables (across multiple languages, not just in Rust) is to start with `is_`. For instance `is_compressed` and `is_unsigned`. It works well with `if` statements (e.g. `if load.is_compressed { ... }` and it makes boolean easy to identify when reading the code.